pyat/at/lattice/variable_elements.py

```diff
@@ -1,141 +1,280 @@
+            amp_aux[key] = lower_case_kwargs.pop("amplitude" + key.lower(), None)
+            if amp_aux[key] is not None:
+                all_amplitudes_are_none = False
+        if all_amplitudes_are_none:
+            raise AtError("Please provide at least one amplitude for A or B")
+        return amp_aux
+
+    def _set_amplitude(self, amplitude: float or _array or None) -> _array:
+        if np.isscalar(amplitude):
+            amplitude = [amplitude]
+        return np.asarray(amplitude)
 
-    def _setmaxorder(self, ampa, ampb):
+    def _getmaxorder(self, ampa: np.ndarray, ampb: np.ndarray) -> int:
         mxa, mxb = 0, 0
         if ampa is not None:
-            mxa = numpy.max(numpy.nonzero(ampa))
+            mxa = np.max(np.append(np.nonzero(ampa), 0))
         if ampb is not None:
-            mxb = numpy.max(numpy.nonzero(ampb))
-        self.MaxOrder = max(mxa, mxb)
-        if ampa is not None:
-            delta = self.MaxOrder - len(ampa)
-            if delta > 0:
-                ampa = numpy.pad(ampa, (0, delta))
-            setattr(self, 'AmplitudeA', ampa)
-        if ampb is not None:
-            delta = self.MaxOrder + 1 - len(ampb)
-            if delta > 0:
-                ampb = numpy.pad(ampb, (0, delta))
-            setattr(self, 'AmplitudeB', ampb)
-
-    def _set_params(self, amplitude, mode, ab, **kwargs):
-        if amplitude is not None:
-            if numpy.isscalar(amplitude):
-                amp = numpy.zeros(self.MaxOrder)
-                amplitude = numpy.append(amp, amplitude)
-            if mode == ACMode.SINE:
-                self._set_sine(ab, **kwargs)
-            if mode == ACMode.ARBITRARY:
-                self._set_arb(ab, **kwargs)
-        return amplitude
-
-    def _set_sine(self, ab, **kwargs):
-        frequency = kwargs.pop('Frequency' + ab, None)
-        phase = kwargs.pop('Phase' + ab, 0)
-        assert frequency is not None, \
-            'Please provide a value for Frequency' + ab
-        setattr(self, 'Frequency' + ab, frequency)
-        setattr(self, 'Phase' + ab, phase)
-
-    def _set_arb(self, ab, **kwargs):
-        func = kwargs.pop('Func' + ab, None)
+            mxb = np.max(np.append(np.nonzero(ampb), 0))
+        return max(mxa, mxb)
+
+    def _set_mode(
+        self, mode: int, a_b: str, **kwargs: dict[str, any]
+    ) -> dict[str, any]:
+        if mode == ACMode.WHITENOISE:
+            kwargs = self._set_white_noise(a_b, **kwargs)
+        if mode == ACMode.SINE:
+            kwargs = self._set_sine(a_b, **kwargs)
+        if mode == ACMode.ARBITRARY:
+            kwargs = self._set_arb(a_b, **kwargs)
+        return kwargs
+
+    def _set_white_noise(self, a_b: str, **kwargs: dict[str, any]) -> dict[str, any]:
+        if "Mean" + a_b not in kwargs:
+            kwargs["Mean" + a_b] = kwargs.get("Mean" + a_b, 0)
+        if "Std" not in kwargs:
+            kwargs["Std" + a_b] = kwargs.get("Std" + a_b, 1)
+        return kwargs
```

Comment:
I think that `Mean` and `Std` do not bring anything : the static part "Amplitude*Mean" can be provided by a standard `ThinMultipole` element, and for the dynamic part, `Amplitude` and `Std` are redundant. I would simplify both python and C b removing these attributes: `Amplitude` is sufficient.